NightscoutServiceKit/RemoteCommands/V1/Notifications/BolusRemoteNotification.swift

```diff
@@ -10,35 +10,34 @@ import Foundation
 import LoopKit
 
 public struct BolusRemoteNotification: RemoteNotification, Codable {
-    
+
     public let amount: Double
     public let remoteAddress: String
     public let expiration: Date?
     public let sentAt: Date?
     public let otp: String
+    public let enteredBy: String?
     
     enum CodingKeys: String, CodingKey {
         case remoteAddress = "remote-address"
         case amount = "bolus-entry"
         case expiration = "expiration"
         case sentAt = "sent-at"
         case otp = "otp"
-    }
-    
-    func toRemoteCommand(otpManager: OTPManager, commandSource: RemoteCommandSource) -> NightscoutRemoteCommand {
-        let expirationValidator = ExpirationValidator(expiration: expiration)
-        let otpValidator = OTPValidator(sentAt: sentAt, otp: otp, otpManager: otpManager)
-        return NightscoutRemoteCommand(id: id,
-                                       action: toRemoteAction(),
-                                       validators: [expirationValidator, otpValidator],
-                                       commandSource: commandSource
-        )
+        case enteredBy = "entered-by"
     }
     
     func toRemoteAction() -> Action {
         return .bolusEntry(BolusAction(amountInUnits: amount))
     }
     
```

Comment:
The validate method on each notification will do both the OTP validation, if required, and validate the expiration date.